### DISCUSSION TITLE
Readonly variables inside method scopes

### DISCUSSION THREAD

**Discussion by osamakawish** [dotnet/csharplang#4250] [Language Ideas] [answered]
Basically I'd like something like this, if it's feasible:

```c#
public void DoStuff() {
    readonly int len = Enumarable.Count()
}
```

I can't use `Count()` likely because I'm evaluating a method, and methods are dynamic. But I don't want the `len` variable to change throughout the scope of the method, and it would be nice to express that. Tested the above code and getting the error `the modifier readonly is not valid for this item`.

### Issues
Main issue here is that it's just... really not essential of a feature, but more of a nice addition. Function scopes aren't that large that adding this feature would reduce any bugs, or even easily cause any.

**Comment by HaloFour** [comment, marked as answer]
See: https://github.com/dotnet/csharplang/issues/188
